test/test_helper.rb

```diff
@@ -23,4 +23,27 @@ class ActiveSupport::TestCase
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
   fixtures :all
   # Add more helper methods to be used by all tests here...
+  def setup
+      # Once you have enabled test mode, all requests
+      # to OmniAuth will be short circuited to use the mock authentication hash.
+      # A request to /auth/provider will redirect immediately to /auth/provider/callback.
+      OmniAuth.config.test_mode = true
+    end
+
+    def mock_auth_hash(user)
+      return {
+        provider: user.provider,
+        uid: user.uid,
+        info: {
+          email: user.email,
+          username: user.username
```

Comment:
This doesn't actually match the form of GitHub's response (`username` instead of `nickname`), which is how that bug above slipped through.